### ISSUE TITLE
Function components cannot be given refs

### ISSUE THREAD

**Issue by Mellbourn** [danstepanov/create-expo-stack#239]
When creating a new project, I immediately get a warning on startup

## Expected Behavior

Startup should have no warnings

## Current Behavior

I get a warning:
```
Console warning
Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `SlotClone`.
```

<img width="497" alt="image" src="https://github.com/danstepanov/create-expo-stack/assets/1785727/4add2d60-2993-4652-b88f-3c2c40ae50ca">

Stack output on console
```
Check the render method of `SlotClone`.
    in HeaderButton (at _layout.tsx:27)
    in SlotClone (created by Slot)
    in Slot (at Link.js:133)
    in ExpoRouterLink (at _layout.tsx:26)
    in RCTView (at View.js:116)
```

## Steps to Reproduce

1. npx create-expo-stack
2. I chose: `my-expo-app`, typescript, npm v10.5.0, expo router navigation, drawer+tabs, Tamagui, no authentication, internationalization
```
{
  projectName: 'my-expo-app',
  packages: [
    {
      name: 'expo-router',
      type: 'navigation',
      options: { type: 'drawer + tabs' }
    },
    { name: 'tamagui', type: 'styling' },
    { name: 'i18next', type: 'internationalization' }
  ],
  flags: {
    noGit: false,
    noInstall: false,
    overwrite: false,
    importAlias: true,
    packageManager: 'npm'
  }
}
```
5. `npm run ios`

## Context (Environment)

macOS Sonoma 14.4.1
XCode 15.3

**Comment by roy-sandoval** [comment]
Happens to me too

**Comment by danstepanov** [comment]
Known issue, known solution, I'll resolve this today, sorry fam 😅

**Comment by danstepanov** [comment]
resolved in v2.6.2
